vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,317 @@
+                List<String> spaceAllLink = new ArrayList<>();
+
+                if (spaceToModulelinksList != null) {
+                    spaceToModulelinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getModule() != null
+                                && link.getLink().getModule().getId() != null) {
+                            spaceAllLink.add(link.getLink().getModule().getId());
+                        }
+                    });
+                }
+                if (spaceToSpaceLinksList != null) {
+                    spaceToSpaceLinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getTarget() != null
+                                && link.getLink().getTarget().getId() != null) {
+                            spaceAllLink.add(link.getLink().getTarget().getId());
+                        }
+                    });
+                }
+                if (space.getId() != null) {
+                    spaceLinkMap.put(space.getId(), spaceAllLink);
+                }
+            }
+        }
+    }
+
+    /**
+     * To map each space to all linked space or each space to all linked module
+     * 
+     * @param <T>
+     * @param spaceToAllOtherLinksList passing list of space to space link or space
+     *                                 to module
+     * @param spaceToAllOtherLinksMap  To hold list of space to space link or space
+     *                                 to module corresponding to each space
+     * @param isModule                 boolean variable to identify whether it is
+     *                                 module or not
+     */
+    private <T> void getSpaceToAllOtherLinks(Iterable<T> spaceToAllOtherLinksList,
+            Map<String, List<T>> spaceToAllOtherLinksMap, boolean isModule) {
+
+        for (T link : spaceToAllOtherLinksList) {
+            String spaceId = null;
+            if (isModule) {
+                IModuleLinkDisplay moduleLink = (IModuleLinkDisplay) link;
+
+                if (moduleLink.getLink() != null && moduleLink.getLink().getSpace() != null
+                        && moduleLink.getLink().getSpace().getId() != null) {
+
+                    spaceId = moduleLink.getLink().getSpace().getId();
+                }
+            } else {
+                ISpaceLinkDisplay spaceLink = (ISpaceLinkDisplay) link;
+
+                if (spaceLink.getLink() != null && spaceLink.getLink().getSourceSpace() != null
+                        && spaceLink.getLink().getSourceSpace().getId() != null) {
+
+                    spaceId = spaceLink.getLink().getSourceSpace().getId();
+                }
+            }
+            if (spaceId != null) {
+                if (!spaceToAllOtherLinksMap.containsKey(spaceId)) {
+                    spaceToAllOtherLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToAllOtherLinksMap.get(spaceId).add(link);
+            }
+        }
+    }
```

Comment:
break this into two methods: one for spacelinkdiplays one for modulelinkdisplays; you're not saving a lot of code by having a generic method, it just makes it more confusing.